File: pinyougousellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

//品牌接口
public interface BrandService {
//    List<TbBrand> findAll();//查询所有品牌

    /**
     * 品牌分页
     * @param pageNum     当前页
     * @param pageSize      每页记录数
     * @return
     */
//    PageResult findPage(int pageNum,int pageSize);

    /**
     * 查询
     * @param brand       查询条件的参数
     * @param pageNum     当前页
     * @param pageSize      每页记录数
     * @return
     */
    PageResult findPage(TbBrand brand ,int pageNum,int pageSize);


    void addBrand(TbBrand brand);//添加品牌

    TbBrand findOneById(long id);//根据id查询

    void updateBrand(TbBrand brand);//修改品牌

    void delteBrand(Long[] ids);




}

File: pinyougou-manager-web/src/main/webapp/js/service/brandservice.js
//品牌服务层
app.service("brandService",function ($http) {
    this.findOne=function(id){
        return $http.get('../brand/findOne.do?id='+id)
    };

    this.add=function (entity) {
        return $http.post('../brand/add.do',entity)
    };
    this.update=function (entity) {
        return $http.post('../brand/update.do',entity)
    };
    this.deleteBrands=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    };
    this.search=function(pageNum,pageSize,searchEntity){
        return $http.post('../brand/search.do?pageNum='+pageNum+'&pageSize='+pageSize,searchEntity)
    }
    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }
});
